### PR TITLE
Add attached functions for v5.

### PR DESCRIPTION
**Pull request by coltfred**
Added functions and types for dealing with v5 attached documents.

### CODE REVIEW

File: src/v5/attached.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    if encoded_edek.len() > u16::MAX as usize {
+        Err(Error::HeaderLengthOverflow(encoded_edek.len() as u64))
+    } else {
+        let len = encoded_edek.len() as u16;
+
+        let result = [
+            key_id_header_bytes.as_ref(),
+            &len.to_be_bytes(),
+            &encoded_edek,
+            &edoc.0, // Note that the edoc is written without a header since it's an attached document.
```
Comment (giarc3) [inline src/v5/attached.rs]: By "without a header" do you mean "0IRON"? It's a bit confusing because this does have a V4DocumentHeader right before it.

File: src/v5/attached.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+#[derive(Debug, PartialEq)]
+pub struct AttachedDocument {
+    pub key_id_header: KeyIdHeader,
+    pub edek: V4DocumentHeader,
+    pub edoc: IvAndCiphertext,
+}
+
+/// Construct an IronCore attached EDOC from the constituent parts.
+pub fn encode_attached_edoc(
```
Comment (giarc3) [inline src/v5/attached.rs]: It feels like this could just be a function on `AttachedDocument` with an "into bytes" type of name

File: src/v5/attached.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            &encoded_edek,
+            &edoc.0, // Note that the edoc is written without a header since it's an attached document.
+        ]
+        .concat();
+        Ok(result.into())
+    }
+}
+
+/// Breaks apart an attached edoc into its parts.
+pub fn decode_attached_edoc(b: Bytes) -> Result<AttachedDocument> {
```
Comment (giarc3) [inline src/v5/attached.rs]: Similarly, this feels like a TryInto from Bytes to AttachedDocument

File: src/v5/attached.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    let (key_id_header, mut attached_document_with_header) =
+        key_id_header::decode_version_prefixed_value(b)?;
+    if attached_document_with_header.len() > 2 {
+        let len = attached_document_with_header.get_u16();
```
Comment (giarc3) [inline src/v5/attached.rs]: 1. Maybe `header_len` as a name to indicate that it's just the V4DocumentHeader length
2. Can you comment here that this advances the byte buffer by 2 bytes? I forgot that and was confused for a bit (or byte)

File: src/v5/attached.rs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        if attached_document_with_header.len() > len as usize {
+            let header_bytes = attached_document_with_header.split_to(len as usize);
+            let edek = protobuf::Message::parse_from_bytes(&header_bytes[..])
+                .map_err(|e| Error::HeaderParseErr(e.to_string()))?;
+            Ok(AttachedDocument {
+                key_id_header,
+                edek,
+                edoc: IvAndCiphertext(attached_document_with_header),
```
Comment (giarc3) [inline src/v5/attached.rs]: Sad that the name `attached_document_with_header` is not true when it gets here, but it is what it is.

File: src/v5/attached.rs [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            })
+        } else {
+            Err(Error::HeaderParseErr("Edek length too long.".to_string()))
```
Comment (giarc3) [inline src/v5/attached.rs]: I would read this error to mean "the EDEK you passed in was too long", but it's meant as "The EDEK you passed in was too short based on the length bytes". Obviously mine is too wordy, but maybe some wordsmithing to clarify

File: src/v5/attached.rs [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    AttachedDocument {
+        key_id_header,
+        edek,
+        edoc,
+    }: &AttachedDocument,
+) -> Result<Bytes> {
+    let key_id_header_bytes = key_id_header.write_to_bytes();
+    let encoded_edek = edek.write_to_bytes().expect("Writing to bytes is safe");
+    if encoded_edek.len() > u16::MAX as usize {
+        Err(Error::HeaderLengthOverflow(encoded_edek.len() as u64))
```
Comment (giarc3) [inline src/v5/attached.rs]: I don't think I did this check in other modules...

**Review by giarc3 — CHANGES_REQUESTED** [review]
Mostly just style/comment suggestions

File: src/v5/attached.rs (hunk 2 above)
Comment (coltfred) [reply src/v5/attached.rs]: I did this because it mirrors how the v4 documents module is setup, but I did move it and we can see how it feels.

File: src/v5/attached.rs (hunk 7 above)
Comment (coltfred) [reply src/v5/attached.rs]: It's checked in v4 attached and v5 attached. Those are the only 2 that use this style format so I think it's ok. What other place did you think we should check this?

File: src/v5/attached.rs (hunk 1 above)
Comment (coltfred) [reply src/v5/attached.rs]: Updated the comment.

File: src/v5/attached.rs (hunk 3 above)
Comment (coltfred) [reply src/v5/attached.rs]: 👍

File: src/v5/attached.rs (hunk 4 above)
Comment (coltfred) [reply src/v5/attached.rs]: I chose `edek_len`. Lmk if you think that's better.

File: src/v5/attached.rs (hunk 6 above)
Comment (coltfred) [reply src/v5/attached.rs]: I don't think it's too long.

**Review by giarc3 — APPROVED** [review]
